### ISSUE TITLE
Why ec delay multiplier must be non-zero?

### ISSUE THREAD

**Issue by masih** [filecoin-project/go-f3#783]
It is unclear why this condition exists. Looking at the code, zero can also be a valid value. No?

https://github.com/filecoin-project/go-f3/blob/d4f080decf5cd19c037b736263aedbf589faf8bf/manifest/manifest.go#L172

**Comment by Stebalien** [comment]
If that's zero, https://github.com/filecoin-project/go-f3/blob/9c8c8523fc6c2a0168ed2a96e8ff04b3616aeb54/host.go#L346 becomes zero.

Honestly, I think 1 is the sane minimum in general. Unless I'm missing something.

**Comment by masih** [comment]
This ticket was captured as part of running a passive testing that was attempting to minimise the wait time between instances. Specifically to get `ecDelay` to indeed be zero. Because, we will always wait for `HeadLookback` number of epochs anyway.

My question remains: Why `ecDelay` of zero is invalid?

In terms of importance, maybe this does not matter anymore. Or at least it will not matter once transport is made more efficient such that the actual delta is a value closer to the default of 3 seconds.

**Comment by Kubuxu** [comment]
If we are in a caught-up regime, if ECDelay is zero, we will start a new instance before new tipsets become available.

**Comment by masih** [comment]
> we will start a new instance before new tipsets become available

Why? Headlookback should avoid this already?

**Comment by masih** [comment]
Right, but we take the [max of 0 and len-headlookback](https://github.com/filecoin-project/go-f3/blob/be650d0b02da81e71bd28f9500942071ecaf9701/consensus_inputs.go#L137) when collecting chain. If we are _too close_ to the head then, would that not end up delaying the instance anyway? i.e. collecting base as the chain which would eventually back off from the start of next instance?

**Comment by Kubuxu** [comment]
It will be missing one or two epochs, I think two.

**Comment by BigLep** [comment]
@masih : is this a cleanup/improvement you would try and squeeze in before nv25 or is it realistically a potential post f3 activation item?
